From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Moritz Sanft <58110325+msanft@users.noreply.github.com>
Date: Thu, 21 Nov 2024 14:36:23 +0100
Subject: [PATCH] hw/x86: load initrd to static address

For TDX RTMRs to be predictable regardless of VM memory size, we need to
load the initrd to a static address, so no dynamic value ends up in the
mapped kernel image.

Without setting this to a static address, the address the initrd is mapped to
which depend not only on the size of the initrd, but also on the memory space
of the guest, this is not viable for Contrast's reference-value-based attestation
approach.

The actual value of the static address does not matter, since OVMF fills the
corresponding Linux header field with the correct address after measuring the
kernel.

Signed-off-by: Moritz Sanft <58110325+msanft@users.noreply.github.com>
---
 hw/i386/x86.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hw/i386/x86.c b/hw/i386/x86.c
index 504575abfa98bc25e498e219a2d58d8d31e5feaa..ec109d0d24256b11ea0f5ad2e4ea57d69aebc915 100644
--- a/hw/i386/x86.c
+++ b/hw/i386/x86.c
@@ -1042,7 +1042,7 @@ void x86_load_linux(X86MachineState *x86ms,
         sev_load_ctx.initrd_data = initrd_data;
         sev_load_ctx.initrd_size = initrd_size;

-        stl_p(header + 0x218, initrd_addr);
+        stl_p(header + 0x218, 0x80000000); // Placeholder, will be overwritten by OVMF.
         stl_p(header + 0x21c, initrd_size);
     }
